src/main/java/com/velox/sloan/CmoSampleIdRegeneratorPlugin.java

```diff
@@ -64,9 +70,10 @@ protected PluginResult run() throws Throwable {
             init();
             List<DataRecord> sampleCMOInfoRecords = activeTask.getAttachedDataRecords("SampleCMOInfoRecords", user);
             logDebug("Regenerating CMO Sample Ids for all attached Sample CMO Info Records");
-            for (DataRecord sampleCMOInfoRecord : sampleCMOInfoRecords) {
-                regenerateCMOSampleId(sampleCMOInfoRecord);
-            }
+            Map<String, CmoInfoRecord> igoId2CmoInfoRecords = getCmoInfoRecords(sampleCMOInfoRecords);
+
+            fillInNewCmoSampleIds(igoId2CmoInfoRecords);
```

Comment:
I'm a bit worried about this throw here. I worry that the errors don't really indicate the fix. Normally this isn't an issue, because they tell you and you look at the logs.  But now there aren't people to look at the logs. Will it be solvable?